### ISSUE TITLE
Fix request: AvoidUnguardedAssignmentToNonFinalFieldsInSharedObjects false positive with static class

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#520]
```java
@RestController
public class Foo {

    public enum Type {
        SMS, EMAIL
    }

    public String getProviders(Type type) {
        String contact = null;
        switch (type) {
            case SMS:
                contact = "1234567890"; // <- false postive
                break;
            case EMAIL:
                contact = "me@example.com"; // <- false positive
                break;
        }
        return contact;
    }

    public static class Info {
        private boolean isDefault;
        private String contact; // <- causes false positive
    }
}
```
